src/design-system/components/Inline/Inline.tsx

```diff
@@ -39,20 +50,40 @@ export function Inline({
         alignVertical ? alignVerticalToFlexAlign[alignVertical] : undefined
       }
       flexDirection="row"
-      flexWrap="wrap"
+      flexWrap={wrap ? 'wrap' : undefined}
       justifyContent={
         alignHorizontal
           ? alignHorizontalToFlexAlign[alignHorizontal]
           : undefined
       }
-      marginRight={horizontalSpace ? negateSpace(horizontalSpace) : undefined}
-      marginTop={verticalSpace ? negateSpace(verticalSpace) : undefined}
+      marginRight={
+        wrap && horizontalSpace ? negateSpace(horizontalSpace) : undefined
+      }
+      marginTop={wrap && verticalSpace ? negateSpace(verticalSpace) : undefined}
     >
-      {Children.map(flattenChildren(children), child => (
-        <Box paddingRight={horizontalSpace} paddingTop={verticalSpace}>
-          {child}
-        </Box>
-      ))}
+      {Children.map(flattenChildren(children), (child, index) => {
+        if (wrap) {
+          return (
+            <Box paddingRight={horizontalSpace} paddingTop={verticalSpace}>
+              {child}
+            </Box>
+          );
+        }
+
+        const isLastChild = index === flattenChildren(children).length - 1;
```

Comment:
This call to `flattenChildren(children)` should be hoisted out to a variable so it's only called once per render.